src/plugins/vis_type_table/public/components/table_vis_grid_columns.tsx

```diff
@@ -15,14 +15,15 @@ export const getDataGridColumns = (
   event: IInterpreterRenderHandlers['event'],
   columnWidths: ColumnWidth[]
 ) => {
+  const formattedColumnIds = table.formattedColumns.map((column) => column.id);
```

Comment:
Nit: I might make it a `Set` here
```suggestion
  const formattedColumnIds = new Set(table.formattedColumns.map((column) => column.id));
```

Reply:
Spoke to Josh. This will only be filtering a handful of `formattedColumns` and as such, a `Set` could be detrimental.

Reply:
Thanks @AMoo-Miki , not sure if I get it, why `Set` could be detrimental?
PS: I don't insist on that, this is a very minor thing :)

Reply:
Even though looksups are cheaper with `Set`, the act of instantializing and populating are expensive overheads. I created two examples to show this better:
[Array vs Set - Small Batch](https://perf.link/#eyJpZCI6IjI4dGU3eXZqeHl4IiwidGl0bGUiOiJTZXQgdnMgQXJyYXkgLSBTbWFsbCBCYXRjaCIsImJlZm9yZSI6ImNvbnN0IGhleXN0YWNrU21hbGwgPSAnTG9yZW0gaXBzdW0gZG9sb3Igc2l0IGFtZXQgY29uc2VjdGV0dXIgZG9sb3Igc2l0IE51bGxhIGlhY3VsaXMganVzdG8gbG9yZW0gZWZmaWNpdHVyIFBlbGxlbnRlc3F1ZSBzdXNjaXBpdCcuc3BsaXQoJyAnKTtcbmNvbnN0IG5lZWRsZXNTbWFsbCA9ICdMb3JlbSBpcHN1bSBkb2xvciBzaXQgYW1ldCBjb25zZWN0ZXR1ciBkb2xvciBzaXQgUGVsbGVudGVzcXVlIHN1c2NpcGl0Jy5zcGxpdCgnICcpOyIsInRlc3RzIjpbeyJuYW1lIjoiRmluZCBpdGVtIDQwMCIsImNvZGUiOiJjb25zdCBuZWVkbGVTZXQgPSBuZXcgU2V0KG5lZWRsZXNTbWFsbCk7XG5oZXlzdGFja1NtYWxsLmZpbHRlcihpdGVtID0%2BIG5lZWRsZVNldC5oYXMoaXRlbSkpOyIsInJ1bnMiOlsxNzMwMDAsMTA0MDAwLDIwMDAsMTQ5MDAwLDEwMDAsMTAwMDAwLDEwMDAwMCwxNTAwMCwxNjEwMDAsMzgwMDAsOTAwMDAsOTgwMDAsMjkwMDAsMTk2MDAwLDk4MDAwLDUwMDAsNTAwMCwxODkwMDAsODQwMDAsMTE4MDAwLDM4MDAwLDQ0MDAwLDExOTAwMCwyMjAwMCwxMTAwMDAsOTgwMDAsNjAwMCwzMDAwLDEwMDAsNDYwMDAsMTAwMDAwLDExMDAwLDc4MDAwLDIwMDAsOTAwMDAsMzYwMDAsMTAwMDAwLDE4MDAwLDkwMDAwLDk4MDAwLDEyMzAwMCw5MDAwMCw3MzAwMCw4OTAwMCw3NzAwMCwxNjEwMDAsOTAwMDAsNTMwMDAsNTYwMDAsMTAwMCwxMDAwLDg0MDAwLDUyMDAwLDEyNzAwMCwxMjAwMCw3OTAwMCwxMjAwMCwxMTgwMDAsNzAwMCwxMDAwLDMyNzAwMCwyOTAwMCw4MzAwMCw2MDAwLDY2MDAwLDE2MDAwMCwxMDAwMCwxMDAwLDQ3MDAwLDkwMDAwLDUwMDAsMTQyMDAwLDEwMDAsMjAwMCwxMDIwMDAsNTMwMDAsMTAwMCwxMDAwLDIyMDAwLDEwMDAsMTI3MDAwLDM4MDAwLDEwMDAwLDQwMTAwMCwxOTcwMDAsMTEwMDAsMTUwMDAsMTI4MDAwLDEwMTAwMCw4NDAwMCwxNzEwMDAsMTE4MDAwLDI5MjAwMCw3MzAwMCwxMjAwMDAsMTg3MDAwLDcxMDAwLDE2MDAwLDU0MDAwLDk3MDAwXSwib3BzIjo3NjMxMH0seyJuYW1lIjoiRmluZCBpdGVtIDgwMCIsImNvZGUiOiJoZXlzdGFja1NtYWxsLmZpbHRlcihpdGVtID0%2BIG5lZWRsZXNTbWFsbC5pbmNsdWRlcyhpdGVtKSk7IiwicnVucyI6WzI4ODAwMCw5ODAwMCwyMDAwLDI0NzAwMCw2MDYwMDAsMTAwMDAwLDE5OTAwMCwxMDAwLDMzODAwMCwxNDEwMDAsMjA4MDAwLDEwMDAwMCwzNTAwMCw3MDAwLDE3OTAwMCwxMDAwLDYwMDAsMjIzMDAwLDEwMDAwMCwxMDAwLDI2MDAwLDEwMDAwMCwyMTQwMDAsNDIwMDAsMTAwMCwxNDkwMDAsMzA3MDAwLDIxNjAwMCw5MDAwLDcwMDAsMTE3MDAwLDEwMDAsOTgwMDAsOTgwMDAsMTI5MDAwLDE5NzAwMCwxNTMwMDAsMTAwMCwxNDYwMDAsMTg5MDAwLDM0NTAwMCwzMTYwMDAsMTE0MDAwLDIwMDAsMTAwMDAwLDIzOTAwMCwxNjcwMDAsOTAwMCwxNDUwMDAsMTAwMDAwLDIwMDAsMTAzMDAwLDg5MDAwLDEwMDAwMCw3MDAwLDEwMDAsMjUwMDAsMjg0MDAwLDEwMDAsMTEwMDAsNjc3MDAwLDEyMDAwLDE0MTAwMCwxMDAwLDEzMDAwMCwyNTgwMDAsODMwMDAsMjAwMCw0MzAwMCw5MDAwMCwzNTIwMDAsMjk5MDAwLDIxMDAwLDEyMDAwLDkwMDAwLDEwMTAwMCwxMzgwMDAsMTAwMCw3MDAwMCwyNzkwMDAsMjU0MDAwLDU5MDAwLDEwNTAwMCw1NDUwMDAsNjExMDAwLDExMDAwLDUzMDAwLDIzMDAwMCwyNTIwMDAsMTA1MDAwLDI1NTAwMCwzNDYwMDAsNTE4MDAwLDEwMDAwMCwxODcwMDAsMjg5MDAwLDEwMDAwMCwyOTAwMCwxMzkwMDAsMTM2MDAwXSwib3BzIjoxNDM2NDB9XSwidXBkYXRlZCI6IjIwMjMtMTItMjFUMTc6NDc6NTkuMzQxWiJ9): When you have small amount of data, using a `Set` would be less performant because of the overhead.
[Array vs Set - Large Batch](url): With larger data, the lookup is what takes the most amount of time which give us beter performance using `Set`.

Reply:
Looks like you save 1/1000000 seconds.